File: e-commerce/frontend/src/app/shared/models/reports.ts
export interface SalesSummary {
  total_revenue: number;
  total_orders: number;
  average_order_value: number;
  currency: string;
}

export interface ProductStats {
  total_products: number;
  low_inventory_count: number;
  out_of_stock_count: number;
}

export interface OrderStats {
  total_orders: number;
  pending_orders: number;
  confirmed_orders: number;
  cancelled_orders: number;
}

export interface RevenueByPeriod {
  period: string;
  revenue: number;
  order_count: number;
  currency: string;
}

export interface TopProduct {
  product_id: string;
  product_name: string;
  sku: string;
  total_quantity_sold: number;
  total_revenue: number;
  currency: string;
}

export interface DashboardResponse {
  sales_summary: SalesSummary;
  product_stats: ProductStats;
  order_stats: OrderStats;
  revenue_by_period: RevenueByPeriod[];
  top_products: TopProduct[];
  period_start: string;
  period_end: string;
}

File: e-commerce/backend/app/schemas/reports.py
"""Reports and analytics schemas."""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class SalesSummary(BaseModel):
    """Sales summary statistics."""

    total_revenue: Decimal
    total_orders: int
    average_order_value: Decimal
    currency: str


class ProductStats(BaseModel):
    """Product statistics."""

    total_products: int
    low_inventory_count: int
    out_of_stock_count: int


class OrderStats(BaseModel):
    """Order statistics."""

    total_orders: int
    pending_orders: int
    confirmed_orders: int
    cancelled_orders: int


class RevenueByPeriod(BaseModel):
    """Revenue breakdown by time period."""

    period: str  # e.g., "2024-01", "2024-01-15"
    revenue: Decimal
    order_count: int
    currency: str


class TopProduct(BaseModel):
    """Top selling product."""

    product_id: UUID
    product_name: str
    sku: str
    total_quantity_sold: int
    total_revenue: Decimal
    currency: str


class DashboardResponse(BaseModel):
    """Complete dashboard data."""

    sales_summary: SalesSummary
    product_stats: ProductStats
    order_stats: OrderStats
    revenue_by_period: list[RevenueByPeriod]
    top_products: list[TopProduct]
    period_start: datetime
    period_end: datetime

